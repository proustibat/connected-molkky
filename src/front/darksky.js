import React from 'react';
import ReactDOM from 'react-dom';
import DarkSky from './pages/DarkSky';

const app = document.getElementById( 'root' );

ReactDOM.hydrate( <DarkSky {...window.__INITIAL_PROPS__} />, app );
