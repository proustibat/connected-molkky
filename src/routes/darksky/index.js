import Darksky from '@pages/Darksky';
import React from 'react';
import express from 'express';
import get from 'lodash/get';
import getWeather from '@services/getWeather';
import omit from 'lodash/omit';
import renderTemplate from '@root/renderTemplate';

const router = express.Router();

router.get('/', async (req, res) => {
  const weather = await getWeather().catch((err) => res.status(500).send(`Something broke! ${err}`));
  if (weather) {
    const data = {
      title: 'Weather',
      ...omit(weather, ['currently.icon', 'currently.summary', 'currently.time']),
      time: new Date(get(weather, 'currently.time')),
      icon: get(weather, 'currently.icon'),
      summary: get(weather, 'currently.summary'),
    };
    res.writeHead(200, { 'Content-Type': 'text/html' });
    res.end(renderTemplate(<Darksky {...data} />));
  }
});

module.exports = router;
