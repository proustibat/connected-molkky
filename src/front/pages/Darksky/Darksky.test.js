import React from 'react';
import { shallow } from 'enzyme';
import {initProps} from '../../../fixtures/darksky';
import Darksky from './index';

describe('Page Darksky', () => {
    it('should render the page correctly', () => {
        // Given / When
        const component = shallow(<Darksky {...initProps} />);

        // Then
        expect(component).toHaveLength(1);
        expect(component).toMatchSnapshot();
    });
});