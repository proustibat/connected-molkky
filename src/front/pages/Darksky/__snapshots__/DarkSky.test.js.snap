// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Darksky should render the page correctly 1`] = `
<div
  className="section no-pad-bot"
>
  <div
    className="container"
  >
    <h1>
      <i
        className="material-icons medium"
      >
        language
      </i>
      <span>
        Europe/Paris
      </span>
    </h1>
    <h2>
      <span>
        Ciel Nuageux
      </span>
      <DarkSkyIcon
        autoplay={true}
        color="black"
        icon="partly-cloudy-day"
      />
    </h2>
    <h3>
      <i
        className="material-icons small"
      >
        location_on
      </i>
      <span>
        48.8534
        , 
        2.3488
      </span>
    </h3>
    <p>
      (
      Mon Jan 19 1970 05:15:53 GMT+0100 (Central European Standard Time)
      )
    </p>
    <List
      elements={
        Array [
          "precipIntensity: 0.2473",
          "precipProbability: 0.34",
          "precipType: rain",
          "temperature: 15.08",
          "apparentTemperature: 15.08",
          "dewPoint: 12.13",
          "humidity: 0.83",
          "pressure: 1012.3",
          "windSpeed: 3.59",
          "windGust: 8.3",
          "windBearing: 226",
          "cloudCover: 0.8",
          "uvIndex: 0",
          "visibility: 10.539",
          "ozone: 249.3",
        ]
      }
    />
  </div>
</div>
`;
